plugins/octant/cmd/antrea-octant-plugin/traceflow.go

```diff
@@ -88,292 +95,236 @@ func getDstType(tf *crdv1alpha1.Traceflow) string {
+					IP: src,
 				}
 			}
+			sourceName = src
 		}
-		log.Printf("Get user input successfully, traceflow: %+v", tf)
-		tf, err = p.client.CrdV1alpha1().Traceflows().Create(ctx, tf, v1.CreateOptions{})
+		destination := crdv1alpha1.Destination{}
+		if dstLen == 0 && isSrcPodType {
+			goto ContinueWithoutCheckDst
+		}
+
+		destination, err = checkDestination(request, dst, dstType, dstNamespace, true, isSrcPodType)
 		if err != nil {
-			log.Printf("Failed to create traceflow CRD \"%s\", err: %s", tfName, err)
-			alert := action.CreateAlert(action.AlertTypeError, fmt.Sprintf("Failed to create traceflow CRD, "+
-				"err: %s", err), action.DefaultAlertExpiration)
-			request.DashboardClient.SendAlert(request.Context(), request.ClientID, alert)
 			return nil
 		}
-		log.Printf("Create traceflow CRD \"%s\" successfully, Traceflow Results: %+v", tfName, tf)
-		alert := action.CreateAlert(action.AlertTypeSuccess, fmt.Sprintf("Traceflow \"%s\" is created successfully",
-			tfName), action.DefaultAlertExpiration)
-		request.DashboardClient.SendAlert(request.Context(), request.ClientID, alert)
-		// Automatically delete the traceflow CRD after created for 300s(5min).
-		go func(tfName string) {
-			age := time.Second * 300
-			time.Sleep(age)
-			err := p.client.CrdV1alpha1().Traceflows().Delete(context.Background(), tfName, v1.DeleteOptions{})
-			if err != nil {
-				log.Printf("Failed to delete traceflow CRD \"%s\", err: %s", tfName, err)
-				return
-			}
-			log.Printf("Deleted traceflow CRD \"%s\" successfully after %.0f seconds", tfName, age.Seconds())
-		}(tf.Name)
-		p.lastTf = tf
-		p.graph, err = graphviz.GenGraph(p.lastTf)
+	ContinueWithoutCheckDst:
+		// It is not required for users to input port numbers.
+		hasSrcPort, hasDstPort, srcPort, dstPort := checkPorts(request)
+		hasTimeout, timeout := checkTimeout(request)
+		protocol, err := checkProtocol(request)
+		if err != nil {
+			return nil
+		}
+		// It is not required for users to input port numbers.
+		dropOnlyChecked := false
+		dropOnly, err := request.Payload.StringSlice(dropOnlyCol)
+		if err != nil || len(dropOnly) == 0 {
+			alertPrinter(request, invalidInputMsg+"failed to get dropOnly as string slice: ",
+				"Failed to get dropOnly as string, please check your input and submit again.", nil, err)
+			return nil
+		}
+		if dropOnly[0] == "Yes" {
+			dropOnlyChecked = true
+		}
+
+		// Judge whether the name of traceflow is duplicated.
+		// If it is, then the user created more than one traceflow in one second, which is not allowed.
+		tfName := ""
+		if srcLen == 0 {
+			tfName = "live-dst-" + dst + "-" + time.Now().Format(TIME_FORMAT_YYYYMMDD_HHMMSS)
+		} else if dstLen == 0 {
+			tfName = "live-src-" + sourceName + "-" + time.Now().Format(TIME_FORMAT_YYYYMMDD_HHMMSS)
+		} else {
+			tfName = "live-" + sourceName + "-" + dst + "-" + time.Now().Format(TIME_FORMAT_YYYYMMDD_HHMMSS)
+		}
+		ctx := context.Background()
+		err = p.checkDuplicateTf(tfName, ctx, request)
```

Comment:
Not introduced by this PR, but there is no point to check duplicate name before creating it and it doesn't really help prevent conflict as the same name resource can be created after the check is done.
If it's duplicate, creating it will return `409 Conflict`, and the code can check the error and report duplication.

Reply:
it will actually skip tf creation and return error immediately, so I think it's worthy to do the check instead of posting the data to backend to get conflict error.

Reply:
1. Getting it and creating it are two API calls, how is it worthy when the single create API already does the same check?
2. There is no any lock mechanism to prevent another client from creating the resource with same name between this GET and POST call, the code still needs to handle the create error anyway.

Reply:
ah, Ok, I got your point now, I will check and refine this part. thanks.

Reply:
@tnqn I just updated the code, remove unnecessary check and leave createTfCR() to raise the error during creation.